rs/nns/cmc/src/limiter.rs

```diff
@@ -28,9 +29,35 @@ impl Limiter {
         }
     }
 
+    /// This function clears old events, checks against the provided limit, and returns an
+    /// error if the cycles cannot be added due to exceeding the limit.
+    pub fn check_and_add_cycles(
```

Comment:
Note to reviewer, this logic is moved, but should be the same as its original home in `ensure_balance`